internal/intern/intern.go

```diff
@@ -76,23 +76,38 @@ type Table struct {
 //
 // This function may be called by multiple goroutines concurrently.
 func (t *Table) Intern(s string) ID {
-	if char6, ok := encodeChar6(s); ok {
-		return char6
-	}
-
 	// Fast path for strings that have already been interned. In the common case
 	// all strings are interned, so we can take a read lock to avoid needing
 	// to trap to the scheduler on concurrent access (all calls to Intern() will
 	// still contend mu.readCount, because RLock atomically increments it).
+	if id, ok := t.TryIntern(s); ok {
+		return id
+	}
+
+	// Outline the slow part, to promote inlining of Intern().
+	return t.internSlow(s)
+}
+
+// TryIntern will attempt to intern a string.
+//
+// If interning would result in a new entry being created in the table, returns
+// false. This is useful for e.g. querying an intern-keyed map using a string.
+// Failure to intern the string means that the string has never been seen
+// before, so searching the map will be futile.
+func (t *Table) TryIntern(s string) (ID, bool) {
```

Comment:
The verb "Intern" (like the above method) sounds like "put this element into the intern table". However this method never does anything of the kind.

I think we should instead name it `Query`. So this can be used to query if a particular item is in the table and if so return its ID. And it returns false when the item is not present.

I think that name can make the Go docs describing the operation easier to understand.